src/nidigital/system_tests/test_system_nidigital.py

```diff
@@ -123,6 +119,69 @@ def get_test_file_path(test_name, file_name):
     return os.path.join(test_files_base_dir, test_name, file_name)
 
 
+@pytest.fixture(params=[array.array, numpy.array, list])
+def source_waveform_type(request):
+    if request.param != array.array:
+        pytest.skip('Source waveform types other than array.array are not supported yet (#1132)')
+    return request.param
+
+
+def test_source_waveform_parallel_site_unique(multi_instrument_session, source_waveform_type):
+    # Implementation of write_source_waveform_site_unique() uses
+    # memoryview of array.array, which is not supported in Python 2.x
+    if sys.version_info.major >= 3:
+        test_name = test_source_waveform_parallel_site_unique.__name__
+        configure_session(multi_instrument_session, test_name)
+
+        multi_instrument_session.load_pattern(get_test_file_path(test_name, 'pattern.digipat'))
+
+        num_samples = 256
+        multi_instrument_session.write_sequencer_register(reg='reg0', value=num_samples)
+
+        multi_instrument_session.create_source_waveform_parallel(
+            pin_list='LowPins',
+            waveform_name='src_wfm',
+            data_mapping=2601)
+
+        if source_waveform_type == array.array:
+            source_waveform = {
+                1: array.array('L', [i for i in range(num_samples)]),
+                0: array.array('L', [i for i in reversed(range(num_samples))])}
+        elif source_waveform_type == numpy.array:
+            source_waveform = {
+                1: numpy.array([i for i in range(num_samples)]),
+                0: numpy.array([i for i in reversed(range(num_samples))])}
+        elif source_waveform_type == list:
+            source_waveform = {
+                1: [i for i in range(num_samples)],
+                0: [i for i in reversed(range(num_samples))]}
+        else:
+            assert False
```

Comment:
Is there a way to include a message with the assertion to indicate why it happened?

Reply:
Updated